src/gg20/crypto_tools/k256_serde.rs

```diff
@@ -226,23 +205,52 @@ mod tests {
 
     #[test]
     fn basic_round_trip() {
+        // scalar
         let s = Scalar(k256::Scalar::random(rand::thread_rng()));
         let s_serialized = bincode::serialize(&s).unwrap();
+        assert_eq!(s_serialized.len(), 32);
         let s_deserialized = bincode::deserialize(&s_serialized).unwrap();
         assert_eq!(s, s_deserialized);
 
+        // affine point
         let a = AffinePoint(
             (k256::AffinePoint::generator() * k256::Scalar::random(rand::thread_rng())).to_affine(),
         );
         let a_serialized = bincode::serialize(&a).unwrap();
         let a_deserialized = bincode::deserialize(&a_serialized).unwrap();
         assert_eq!(a, a_deserialized);
 
+        // projective point
         let p = ProjectivePoint(
             k256::ProjectivePoint::generator() * k256::Scalar::random(rand::thread_rng()),
         );
         let p_serialized = bincode::serialize(&p).unwrap();
         let p_deserialized = bincode::deserialize(&p_serialized).unwrap();
         assert_eq!(p, p_deserialized);
     }
+
+    #[test]
+    fn scalar_deserialization_fail() {
+        // test too few bytes
+        let s = Scalar(k256::Scalar::random(rand::thread_rng()));
+        let mut too_few_bytes = bincode::serialize(&s).unwrap();
+        too_few_bytes.pop();
+        bincode::deserialize::<Scalar>(&too_few_bytes).unwrap_err();
+
+        // test too many bytes
+        // TODO bincode ignores extra bytes by default
```

Comment:
If you don't want dependence on the `sdk` to use our custom `serialize/deserialize`, you can do the following directly in tests:
```rust
let options = bincode::DefaultOptions::new();
options.serialize(...)
```
(the default differs from the custom in that it uses BE instead of LE).

Alternatively, we can move our custom `serialize/deserialize` in `collections`?